libraries/chain/controller.cpp

```diff
@@ -1575,9 +1590,9 @@ struct controller_impl {
             trx_context.exec();
             trx_context.finalize(); // Automatically rounds up network and CPU usage in trace and bills payers if successful
 
-            auto restore = make_block_restore_point();
+            auto restore = make_block_restore_point( trx->is_read_only() );
 
-            if (!trx->implicit()) {
+            if (!trx->implicit() && !trx->is_read_only()) {
```

Comment:
I think you want `!trx->is_transient()`. We do not want to put receipts of compute transactions in the pending block either.

We do want to update `trx->billed_cpu_time_us` as it is logged in producer plugin. Maybe we just do that for all trxs including implicit.

Reply:
Thanks. Done.

Reply:
Sorry. In our meeting today I think we decided to leave it as you had it with only skip for read-only. For compute transaction we want to measure the time for creating the receipt.

Reply:
Oh yes, we would make compute_transaction as close as real one. Will do it only for read-only.